src/EditorFeatures/Core/CommandHandlers/InteractivePasteCommandHandler.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+extern alias InteractiveWindow;
+
+using System;
+using System.ComponentModel.Composition;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Windows;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.CodeAnalysis.Editor.Commands;
+
+namespace Microsoft.CodeAnalysis.Editor.CommandHandlers
+{
+    [ExportCommandHandler("InteractivePasteCommandHandler", ContentTypeNames.RoslynContentType)]
+    [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
```

Comment:
Should we enforce ordering between this and the other things listening to Paste? For example, there's some "interesting" paste handling in the RenameCommandHandler. If we get the ordering wrong, you may try to write to a readonly part of the buffer, etc. Let's figure out the correct ordering between all paste handlers.